rcl/src/rcl/arguments.c

```diff
@@ -1597,17 +1662,72 @@ rcl_ret_t
 _rcl_parse_log_level(
   const char * arg,
   rcl_allocator_t allocator,
-  int * log_level)
+  rcl_log_levels_t * log_level)
 {
   RCL_CHECK_ARGUMENT_FOR_NULL(arg, RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ARGUMENT_FOR_NULL(log_level, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(log_level->logger_settings, RCL_RET_INVALID_ARGUMENT);
+
+  size_t separator_len = strlen(RCL_LOG_LEVEL_FLAG_SEPARATOR);
+  rcutils_ret_t ret = RCUTILS_RET_OK;
+  char * name = NULL;
+  char * level = NULL;
+  int logger_level = 0;
+  char * separator = strstr(arg, RCL_LOG_LEVEL_FLAG_SEPARATOR);
+
+  if (separator) {
```

Comment:
Please, use the [lexer](https://github.com/ros2/rcl/blob/f6736211c90067f2d93778ef98c4b47d4f5791ee/rcl/include/rcl/lexer.h).
See for example https://github.com/ros2/rcl/blob/f6736211c90067f2d93778ef98c4b47d4f5791ee/rcl/src/rcl/arguments.c#L1734.

Reply:
Updated